Makefile

```diff
@@ -307,7 +307,7 @@ test: test-deps
 			--format pkgname-and-test-fails \
 			-- \
 				./pkg/... ./utils/... ./cmd/... \
-				$(COVERAGE_OPTS) --tags=unit,allcomponents
+				$(COVERAGE_OPTS) -timeout=20m --tags=unit,allcomponents
```

Comment:
Could you share the name of them?

Increase timeout is not good enough, it may hide some issues.

Reply:
For example, the macos platform unit test has been tried 4 times and finally succeed. The first 3 attempts have different error message, which is more likely to timeout than flaky tests. And I haven't found a specific test consumes too much time.
https://github.com/dapr/dapr/actions/runs/6761791966/job/18385528740?pr=7132
Also lint and intergration test have 20m of timeout.

```
=== FAIL: pkg/placement TestMembershipChangeWorker (11.43s)

DONE 2993 tests, 4 failures in 606.592s
Error: Process completed with exit code 2.
```

```
=== Failed
=== FAIL: pkg/grpc TestCallLocalStream/appchannel_is_not_ready (0.00s)
    api_daprinternal_test.go:125: 
        	Error Trace:	/Users/runner/work/dapr/dapr/pkg/grpc/api_daprinternal_test.go:125
        	Error:      	Received unexpected error:
        	            	EOF
        	Test:       	TestCallLocalStream/appchannel_is_not_ready

=== FAIL: pkg/grpc TestCallLocalStream (0.00s)

DONE 2993 tests, 2 failures in 466.435s
make: *** [test] Error 1
Error: Process completed with exit code 2.
```

```
=== Failed
make: *** [test] Error 1
=== FAIL: pkg/grpc/proxy TestProxySuite/TestPingStream_MultipleThreads (10.01s)
    handler_test.go:313: 
        	Error Trace:	/Users/runner/work/dapr/dapr/pkg/grpc/proxy/handler_test.go:313
        	Error:      	Timed out waiting for proxy to return.
        	Test:       	TestProxySuite/TestPingStream_MultipleThreads

=== FAIL: pkg/grpc/proxy TestProxySuite (21.64s)
    handler_test.go:827: starting grpc.Server at: 127.0.0.1:49281
    handler_test.go:829: starting grpc.Proxy at: 127.0.0.1:49280

DONE 2993 tests, 2 failures in 557.867s
```
and finally success
```
DONE 2993 tests in 513.828s
````

Reply:
I agree with @daixiang0 here, I think we should focus our efforts in getting our unit tests to be less flakey & complete faster. 10m is already a long time for unit tests to complete in the project.

Reply:
I agree on not increasing the timeout for unit tests.

Reply:
Thanks for all the viewpoints. Close this based on the above discussion.